tasks/managed/upload-sbom-to-atlas/upload-sbom-to-atlas.yaml

```
@@ -384,7 +384,8 @@ spec:
 
           echo "Pushing $to_upload to S3."
           if ! curl -X PUT --upload-file "${to_upload}" \
-              --silent --show-error --fail-with-body --retry 5 \
+              --silent --show-error --fail-with-body --retry 10 \
```

I don't really get this. I agree it would help sometimes, but why are we failing so often? Why was 5 not enough (and such a problem we had to turn off failing on this task) but 10 is always enough such that we can turn off `onError: continue`?

---

(maybe you've done a lot of testing and that's how you got to 10, but I think a comment or something should be added explaining this, because just looking at it it looks like we are just trying a higher number to see what happens and maybe we come in next week and go to 50)

---

This is more about fixing the curl bug/quirk, where the request _wasn't_ retried on a 503. The `--retry-all-errors` arg should help here. I guess increasing it to 10 is not required, but if we fail the pipeline on this, we might as well try.

I'd prefer if we kept the `onError: continue` and monitored the pipeline for some time, but AFAIK no automatic monitoring tool exists, and I don't have the permissions to check pipelinerun logs.

If you think ignoring the error and manually monitoring the pipeline would be better here, we can do that.

---

Gotcha, so it would be accurate to say we'd "be surprised" if we still needed even 5 retries due to the new `retry-all-errors` but we are still bumping to 10 just to be safe? I am cool with this change if so

---

Yes, that's the case. S3 _should_ be able to handle thousands of requests per second, if it has time to scale up. I really think this issue is kind of bizarre, hopefully it was just a transient problem.